Make wordpress locale configurable

rogervanwile opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
Currently Wordpress is always installed in the en_US version. With this change it is now possible to determine in which locale Wordpress should be installed. This makes it easy to test non-English plugins.

To use this feature, just add a `locale` to your cypress.json. If no one provided, `en_US` is used.

```
{
    "wp": {
        "locale": "de_DE"
    }
}
```

## Change Log
* Add `locale` as new config setting


## Types of changes (_if applicable_):
- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist (_if applicable_):
- [x] Meets provided linting standards.

liamdefty reviewed (changes requested):
Nice, makes sense and will be a very useful addition.

Just one tiny thing, do you think we could update the validation config schema for this:

https://github.com/bigbite/wp-cypress/blob/master/lib/schemas/config-validation.json

I think we only need to validate it is a string?

rogervanwile commented:
@liamdefty Thanks for your input. I added a config validation rule. I don't know this schema, but I hope I have implemented the right way. I also added a regex pattern.

ampersarnie reviewed (changes requested):
Hi @rogervanwile, sorry it's taken so long for us to pick this back up. I've given this a test and found that the validation doesn't work with some locales and given some additional information. As it's taken a while to get to this I completely understand if you're not able to make the change. If not we can pick up where you leave it.

But thank you for this contribution, it's a valuable one.
